finished lab 7

pdariuslee opened the pull request.

pdariuslee committed:
finished lab 7

RivaD2 commented:
Hey Darius, nice work! I have a question about your Trail constructor? Can you tell me what is happening on lines 139 and 140? I really liked that you added more properties on to your constructor. I just want to know what the .split() means here and what your thought process was.

pdariuslee commented:
> Hey Darius, nice work! I have a question about your Trail constructor? Can you tell me what is happening on lines 139 and 140? I really liked that you added more properties on to your constructor. I just want to know what the .split() means here and what your thought process was.

Line 139 & 140
this.condition_date = jsonObject.conditionDate.split(' ',1);
this.condition_time = jsonObject.conditionDate.split(' ')[1];

Based on Trello, we actually also need the info for date and time for the Trail feature. How I figured where to get that is console logging the object, to see the key-value pairs. 

.conditionDate actually looks kinda like this "Aug 19, 2020 10:05am". And since I need info with just date, and info with just time, I needed to use the split method.

kuya32 commented on server.js:
```diff
@@ -33,6 +33,7 @@ const app = express();
 app.use(cors()); // enables the server to talk to local things
 const GEOCODE_API_KEY = process.env.GEOCODE_API_KEY;
 const WEATHER_API_KEY = process.env.WEATHER_API_KEY;
+const TRAIL_API_KEY = process.env.TRAIL_API_KEY;
```
Putting your API keys to the top is smart, I put mine in each individual function.

kuya32 commented on server.js:
```diff
@@ -104,6 +127,19 @@ function Weather(jsonObject){
+  this.summary = jsonObject.summary;
+  this.trail_url = jsonObject.url;
+  this.conditions = jsonObject.conditionDetails;
+  this.condition_date = jsonObject.conditionDate.split(' ',1);
+  this.condition_time = jsonObject.conditionDate.split(' ')[1];
```
Utilizing the .split method to target the individual pieces of information in the object string is clever.

kuya32 reviewed (commented):
Your code looks great and also was able to pick up some tips along the way!
